src/build/mdn-comments.ts

```diff
@@ -47,66 +45,124 @@ function extractSummary(markdown: string): string {
     return sentenceMatch[0]; // Return the first full sentence
   }
 
-  return normalizedText.split(" ")[0] || ""; // Fallback: first word if no sentence found
+  const firstWord = normalizedText.split(" ")[0];
+  return firstWord || "";
 }
 
-async function getDirectories(dirPath: URL): Promise<URL[]> {
-  try {
-    const entries = await fs.readdir(dirPath, {
-      withFileTypes: true,
-    });
-    return entries
-      .filter((entry) => entry.isDirectory())
-      .map((entry) => new URL(entry.name + "/", dirPath));
-  } catch (error) {
-    console.error("Error reading directories:", error);
-    return [];
+async function walkDirectory(dir: URL): Promise<URL[]> {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
+  const parentDirName = dir.pathname.split("/").filter(Boolean).slice(-1)[0];
+  let results: URL[] = [];
+
+  for (const entry of entries) {
+    if (entry.isDirectory()) {
+      if (entry.name === parentDirName) continue;
+      const subDir = new URL(`${entry.name}/`, dir);
+      results = results.concat(await walkDirectory(subDir));
+    } else if (entry.isFile() && entry.name === "index.md") {
+      results.push(new URL(entry.name, dir));
+    }
   }
+
+  return results;
 }
 
-async function getIndexMdContents(
-  folders: URL[],
-): Promise<{ [key: string]: string }> {
-  const results: { [key: string]: string } = {};
+const types: Record<string, string> = {
+  property: "properties",
+  method: "methods",
+  "landing-page": "ignore",
+  interface: "interfaces",
+  overview: "ignore",
+  event: "events",
+};
 
-  for (const folder of folders) {
-    const indexPath = new URL("index.md", folder);
+function generateTypes(content: string): string[] | undefined {
+  const pageType = content.match(/page-type:\s*["']?([^"'\n]+)["']?/)!;
+  const type = pageType[1].split("-").pop()!;
+  const plural = types[type];
+  if (!plural) return;
 
-    try {
-      const content = await fs.readFile(indexPath, "utf-8");
+  return [plural, type];
+}
 
-      // Improved title extraction
-      const titleMatch = content.match(/title:\s*["']?([^"'\n]+)["']?/);
-      const filename = basename(folder.toString());
-      const title = titleMatch
-        ? titleMatch[1].replace(/ extension$/, "")
-        : filename || "";
+function generateSlug(content: string): string[] {
+  const match = content.match(/slug:\s*["']?([^"'\n]+)["']?/)!;
+  const url = match[1].split(":").pop()!;
+  const parts = url.split("/").slice(2); // skip `/en-US/web/api/...`
```

Comment:
```suggestion
  const normalized = url.endsWith("_static") ? url.slice(0, -7) : url;
  const parts = normalized.split("/").slice(2); // skip `/en-US/web/api/...`
```

This will add some missing comments for static members. (`_static` postfix is added for any static members as MDN people considered any non-static member name will never end with `_static`. Thus it should be safe to remove it without worrying about potential collision)